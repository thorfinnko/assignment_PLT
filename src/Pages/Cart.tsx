import React from 'react';
import { Dimensions, FlatList, StyleSheet, Text, View } from 'react-native';
import { useSelector } from 'react-redux';
import { PriceBottomHeader } from '../Components/PriceBottomHeader';
import { CartCard } from '../Components/CartCard';

function Cart() {
  const { products } = useSelector((state) => state.cart);
  if (products.length > 0) {
    return (
      <View style={{ flex: 1 }}>
        <FlatList
          renderItem={({ item }) => <CartCard item={item} />}
          data={products}
          numColumns={2}
          showsVerticalScrollIndicator={false}
          horizontal={false}
        />
        <View style={{ position: 'absolute', bottom: 0, width: Dimensions.get('screen').width }}>
          <PriceBottomHeader
            price={products.reduce((a, b) => {
              return a + b?.price;
            }, 0) || 0}
          />
        </View>
      </View>
    );
  }

  return (
    <View style={{flex:1, justifyContent:'center', alignItems:'center'}}>
      <Text>No Product Added to cart</Text>
    </View>
  )

}

export default Cart;
