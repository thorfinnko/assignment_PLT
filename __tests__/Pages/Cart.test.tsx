import React from 'react';
import { render, screen, waitFor } from '@testing-library/react-native';
import Cart from '../../src/Pages/Cart';
import { Provider } from 'react-redux';
import createMockStore from '../../__mocks__/mockStore';

const initialState = {
    products: {
        loading: false,
        isError: false,
        products: [],
      },
    cart: {
        products: [],
    },
  };

const store = createMockStore(initialState);

describe('test the cart component', () => {
    it('when there are no products in cart', async () => {
        const {toJSON} = render(
            <Provider store={store}>
                <Cart />
            </Provider>
        );
        expect(screen).toBeTruthy();
        expect(toJSON()).toMatchSnapshot();
        await waitFor(() => {
          expect(screen.getByLabelText('No Product Added to cart')).toBeDefined();
        });
    })
})


