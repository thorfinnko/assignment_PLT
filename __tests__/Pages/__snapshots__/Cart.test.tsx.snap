// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the cart component renders loading screen correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    No Product Added to cart
  </Text>
</View>
`;
